pkg/kubefed2/unjoin.go

```diff
@@ -163,76 +164,25 @@ func UnjoinCluster(hostConfig, clusterConfig *rest.Config, federationNamespace,
 		return err
 	}
 
-	glog.V(2).Infof("Performing preflight checks for unjoin cluster: %s.", unjoiningClusterName)
-	err = performPreflightChecksForUnjoin(clusterClientset, unjoiningClusterName, hostClusterContext, federationNamespace)
-	if err != nil {
-		return err
-	}
-
 	if removeFromRegistry {
-		err = removeFromClusterRegistry(hostConfig, clusterNamespace, clusterConfig.Host, unjoiningClusterName, dryRun)
-		if err != nil {
-			return err
-		}
-	}
-
-	glog.V(2).Infof("Deleting federated cluster resource from namespace: %s for unjoin cluster: %s",
-		federationNamespace, unjoiningClusterName)
-
-	_, err = deleteFederatedClusterAndSecret(hostClientset, fedClientset, federationNamespace,
-		unjoiningClusterName, dryRun)
-	if err != nil {
-		glog.V(2).Infof("Failed to delete federated cluster resource from namespace: %s for unjoin cluster: %s due to: %v",
-			federationNamespace, unjoiningClusterName, err)
-		return err
+		removeFromClusterRegistry(hostConfig, clusterNamespace, clusterConfig.Host, unjoiningClusterName, dryRun)
 	}
 
-	glog.V(2).Infof("Deleted federated cluster resource from namespace: %s for unjoin cluster: %s",
-		federationNamespace, unjoiningClusterName)
-
-	err = deleteRBACResources(hostClientset, clusterClientset,
-		federationNamespace, unjoiningClusterName, hostClusterContext, dryRun)
-	if err != nil {
-		glog.V(2).Infof("Could not delete cluster RBAC resources: %v", err)
-		return err
-	}
+	isSucceed := deleteRBACResources(clusterClientset, federationNamespace, unjoiningClusterName, hostClusterContext, dryRun)
 
 	if unjoiningClusterContext == hostClusterContext {
 		glog.V(2).Infof("Host cluster: %s and unjoin cluster: %s are same, no need to delete federation namespace: %s from unjoin cluster.",
 			hostClusterContext, unjoiningClusterContext, federationNamespace)
 	} else {
-		glog.V(2).Infof("Deleting federation namespace: %s from unjoin cluster: %s",
-			federationNamespace, unjoiningClusterName)
-
-		err = deleteFedNSFromUnjoinCluster(clusterClientset,
-			federationNamespace, unjoiningClusterName, hostClusterContext, dryRun)
+		err = deleteFedNSFromUnjoinCluster(clusterClientset, federationNamespace, unjoiningClusterName, dryRun)
 		if err != nil {
-			glog.V(2).Infof("Could not delete cluster RBAC resources: %v", err)
```

Comment:
How about keep this message as well?

Reply:
If the current strategy of returning an error is maintained, I agree that the error needs to be logged (and via `Errorf`).  But why should `deleteFedNSFromUnjoinCluster` return an error?  It would be more consistent to log the error in the function and return an indication of success instead.